server/src/main/java/org/apache/druid/client/cache/MemcachedCache.java

```diff
@@ -641,14 +641,14 @@ public void close()
 
   public static final int MAX_PREFIX_LENGTH =
       MemcachedClientIF.MAX_KEY_LENGTH
-      - 40 // length of namespace hash
-      - 40 // length of key hash
+      - 64 // length of namespace hash
+      - 64 // length of key hash
       - 2; // length of separators
 
   private static String computeKeyHash(String memcachedPrefix, NamedKey key)
   {
     // hash keys to keep things under 250 characters for memcached
-    return memcachedPrefix + ":" + DigestUtils.sha1Hex(key.namespace) + ":" + DigestUtils.sha1Hex(key.key);
+    return memcachedPrefix + ":" + DigestUtils.sha256Hex(key.namespace) + ":" + DigestUtils.sha256Hex(key.key);
```

Comment:
i will create a test case to get some metric for this comparison.

Reply:
according to following comparison, change to Tiger-192. and Tiger-192 has no major disclosed security vulnerabilities
![image](https://github.com/user-attachments/assets/29eec500-8dae-49c1-bb3b-d33d114993ee)

Reply:
The number look good. Now lets proceed to the testing bit.

The way to test it would be to see if repeated queries are indeed using cache. Can you please try to deploy it on a cluster with at least 2 brokers and 2 historicals and see if things work ?

Reply:
@cryptoe cc: @FrankChen021 
I will deploy and verify it, seems Tiger-192 is risky from GitHub security bot. should we go SHA-256 or continue Tiger-192.

Reply:
@cryptoe @FrankChen021  I have deployed 1 broker, 1 historical and 3 memcached instances, it can work well using following query, i will try 2 brokers and 2 historicals. And i will update here. Just fly back US from China, reply late. 
if need, i could add the docker compose yml file for memcached setup inside this cr.
`select channel, sum(added) from wikipedia group by channel order by sum(added) limit 5
`

Reply:
My opinion is that Tiger-192 is not a widely used alogrithm in the industry, I think SHA256 is better. 
For the performance penalty of SHA256, we can do a test in a cluster for the comparison.